From 42d6e3855d47b3619833509ddd53c490656c3d5d Mon Sep 17 00:00:00 2001
From: Guan Xuetao <gxt@mprc.pku.edu.cn>
Date: Fri, 28 Dec 2012 10:10:08 +0800
Subject: [PATCH 511/641] Revert "UniCore64: Add the definition of
 initrd_start and initrd_end"

This reverts commit 0b447a144b7504bf61d78b9b965b8856b5597af4.
---
 arch/unicore64/include/arch/hwdef-memory.h |    3 ---
 arch/unicore64/mm/init.c                   |   16 ++--------------
 2 files changed, 2 insertions(+), 17 deletions(-)

diff --git a/arch/unicore64/include/arch/hwdef-memory.h b/arch/unicore64/include/arch/hwdef-memory.h
index 5af0769..213eab7 100644
--- a/arch/unicore64/include/arch/hwdef-memory.h
+++ b/arch/unicore64/include/arch/hwdef-memory.h
@@ -72,9 +72,6 @@
 #define UC64_PM_ZIMAGE_STACKTOP		__BC(00000000, 03800000)
 #define UC64_PM_ZIMAGE_HEAP_END		__BC(00000000, 03f00000)
 
-#define UC64_PM_INITRD_START		__BC(00000000, 01000000)
-#define UC64_PM_INITRD_SIZE		__BC(00000000, 01000000)
-
 /*
  * Address space: 00400000 - 00408000
  *
diff --git a/arch/unicore64/mm/init.c b/arch/unicore64/mm/init.c
index b63eafc..9bf230f 100644
--- a/arch/unicore64/mm/init.c
+++ b/arch/unicore64/mm/init.c
@@ -1,13 +1,10 @@
 #include <linux/kernel.h>
 #include <linux/memblock.h>
 #include <linux/bootmem.h>
-#include <linux/initrd.h>
 
 #include <asm/sections.h>
 #include <asm/setup_arch.h>
 
-#include <arch/hwdef-memory.h>
-
 /**
  * mem_init() - release free pages to the buddy allocator
  * Returns the number of pages actually released.
@@ -51,7 +48,8 @@ void free_initmem(void)
 #ifdef CONFIG_BLK_DEV_INITRD
 void free_initrd_mem(unsigned long start, unsigned long end)
 {
-	totalram_pages += __free_area(start, end);
+	/* FIXME */
+	BUG();
 }
 #endif
 
@@ -75,16 +73,6 @@ static void __init memblock_init(void)
 	/* Reserve the page tables. */
 	memblock_reserve(__pa(swapper_pg_dir), 4 * PAGE_SIZE);
 
-#ifdef CONFIG_BLK_DEV_INITRD
-	if (UC64_PM_INITRD_SIZE) {
-		memblock_reserve(UC64_PM_INITRD_START, UC64_PM_INITRD_SIZE);
-
-		/* Now convert initrd to virtual addresses */
-		initrd_start = (unsigned long)__va(UC64_PM_INITRD_START);
-		initrd_end = initrd_start + UC64_PM_INITRD_SIZE;
-	}
-#endif
-
 	memblock_allow_resize();
 	memblock_dump_all();
 }
-- 
1.7.9.5

